### ISSUE TITLE
-Wc++-compat compiler noise on STDERR

### ISSUE THREAD

**Issue by iabyn** [pmqs/Compress-Raw-Zlib#31]
Since Compress-Raw-Zlib was just updated in bleadperl from 2.206 to 2.208, there's a bunch of compiler noise now appearing on STDERR during the bleadperl build. It's coming from in/deflate.c, so I don't know whether it can be tweaked at your end or not.

```
deflate.c: In function ‘Perl_crz_deflateCopy’:
deflate.c:1298:13: warning: request for implicit conversion from ‘void *’ to ‘Perl_crz_Bytef *’ {aka ‘unsigned char *’} not permitted in C++ [-Wc++-compat]
 1298 |     zmemcpy((voidpf)dest, (voidpf)source, sizeof(z_stream));
      |             ^~~~~~~~~~~~
deflate.c:1298:27: warning: request for implicit conversion from ‘void *’ to ‘const Perl_crz_Bytef *’ {aka ‘const unsigned char *’} not permitted in C++ [-Wc++-compat]
 1298 |     zmemcpy((voidpf)dest, (voidpf)source, sizeof(z_stream));
      |                           ^~~~~~~~~~~~~~
deflate.c:1303:13: warning: request for implicit conversion from ‘void *’ to ‘Perl_crz_Bytef *’ {aka ‘unsigned char *’} not permitted in C++ [-Wc++-compat]
 1303 |     zmemcpy((voidpf)ds, (voidpf)ss, sizeof(deflate_state));
      |             ^~~~~~~~~~
deflate.c:1303:25: warning: request for implicit conversion from ‘void *’ to ‘const Perl_crz_Bytef *’ {aka ‘const unsigned char *’} not permitted in C++ [-Wc++-compat]
 1303 |     zmemcpy((voidpf)ds, (voidpf)ss, sizeof(deflate_state));
      |                         ^~~~~~~~~~
deflate.c:1318:13: warning: request for implicit conversion from ‘void *’ to ‘Perl_crz_Bytef *’ {aka ‘unsigned char *’} not permitted in C++ [-Wc++-compat]
 1318 |     zmemcpy((voidpf)ds->prev, (voidpf)ss->prev, ds->w_size * sizeof(Pos));
      |             ^~~~~~~~~~~~~~~~
deflate.c:1318:31: warning: request for implicit conversion from ‘void *’ to ‘const Perl_crz_Bytef *’ {aka ‘const unsigned char *’} not permitted in C++ [-Wc++-compat]
 1318 |     zmemcpy((voidpf)ds->prev, (voidpf)ss->prev, ds->w_size * sizeof(Pos));
      |                               ^~~~~~~~~~~~~~~~
deflate.c:1319:13: warning: request for implicit conversion from ‘void *’ to ‘Perl_crz_Bytef *’ {aka ‘unsigned char *’} not permitted in C++ [-Wc++-compat]
 1319 |     zmemcpy((voidpf)ds->head, (voidpf)ss->head, ds->hash_size * sizeof(Pos));
      |             ^~~~~~~~~~~~~~~~
deflate.c:1319:31: warning: request for implicit conversion from ‘void *’ to ‘const Perl_crz_Bytef *’ {aka ‘const unsigned char *’} not permitted in C++ [-Wc++-compat]
 1319 |     zmemcpy((voidpf)ds->head, (voidpf)ss->head, ds->hash_size * sizeof(Pos));
      |                               ^~~~~~~~~~~~~~~~
inflate.c: In function ‘Perl_crz_inflateCopy’:
inflate.c:1465:13: warning: request for implicit conversion from ‘void *’ to ‘Perl_crz_Bytef *’ {aka ‘unsigned char *’} not permitted in C++ [-Wc++-compat]
 1465 |     zmemcpy((voidpf)dest, (voidpf)source, sizeof(z_stream));
      |             ^~~~~~~~~~~~
inflate.c:1465:27: warning: request for implicit conversion from ‘void *’ to ‘const Perl_crz_Bytef *’ {aka ‘const unsigned char *’} not permitted in C++ [-Wc++-compat]
 1465 |     zmemcpy((voidpf)dest, (voidpf)source, sizeof(z_stream));
      |                           ^~~~~~~~~~~~~~
inflate.c:1466:13: warning: request for implicit conversion from ‘void *’ to ‘Perl_crz_Bytef *’ {aka ‘unsigned char *’} not permitted in C++ [-Wc++-compat]
 1466 |     zmemcpy((voidpf)copy, (voidpf)state, sizeof(struct inflate_state));
      |             ^~~~~~~~~~~~
inflate.c:1466:27: warning: request for implicit conversion from ‘void *’ to ‘const Perl_crz_Bytef *’ {aka ‘const unsigned char *’} not permitted in C++ [-Wc++-compat]
 1466 |     zmemcpy((voidpf)copy, (voidpf)state, sizeof(struct inflate_state));
      |                           ^~~~~~~~~~~~~
```

**Comment by iabyn** [comment]
Further, it's now causing all blead smokers and github continuous integration to fail on builds with -Dcc="g++" and similar.

**Comment by pmqs** [comment]
Thanks. I always forget about the C++ build when I do a release of this module. I'll sort it.

**Comment by pmqs** [comment]
Should be fixed in version 2.209 (just uploaded to CPAN). Ping me if you find any issues.
